src/Agent.Listener/SelfUpdater.cs

```diff
@@ -597,6 +612,40 @@ private async Task UpdateAgentUpdateStateAsync(string currentState)
                 Trace.Info($"Catch exception during report update state, ignore this error and continue auto-update.");
             }
         }
+        /// <summary>
+        /// Verifies authenticode sign of agent assemblies
+        /// </summary>
+        /// <param name="agentFolderPath"></param>
+        /// <returns></returns>
+        private bool VerifyAgentAuthenticode(string agentFolderPath)
+        {
+            if (!Directory.Exists(agentFolderPath))
```

Comment:
Could it make sense to fire an exception in this case?

Reply:
Not really sure since we handle VerifyAgentAuthenticode result further and throw error - for the case when there is something wrong during validation... Added message for this case also.